views/chat.pug

```diff
@@ -5,12 +5,25 @@ html
     h1 Hi #{name}!
     p Go ahead and say something.
 
-    ul
-      each message in messageHistory
-        li= message
+    ul(id='messages')
 
-  form(method='POST' action='/chat')
-    div.form-group
-      label(for='message') Chat:
-      input#name.form-control(type='text', placeholder="What's on your mind?" name='message' value='')
-    button.btn.btn-primary(type='submit') Post
+    form(method='POST' action='/chat')
+      div.form-group
+        label(for='message')
+        input#message.form-control(name='message', type='text', placeholder="What's on your mind?" value='')
+      button.btn.btn-primary(type='submit') Post
+
+    script(src='/socket.io/socket.io.js')
```

Comment:
Where does the client get this resource from?

Reply:
Here is the socket.io documentation on getting the sockets onto the client: https://socket.io/docs/v3/client-installation/

This specific resource is on the server, in the `node_modules` directory, and is served when the client makes a request for the page where this script is located.